### PR TITLE
SONAR-6941 Partition files of scanner report in different folders

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: sonar-scanner-protocol/src/main/java/org/sonar/scanner/protocol/output/FileStructure.java [hunk 1]
```diff
@@ -71,8 +75,30 @@ public File activeRules() {
...
+      result = result.resolve(String.valueOf(c));
+    }
+    return result;
+  }
+
   public File fileFor(Domain domain, int componentRef) {
-    return new File(dir, domain.filePrefix + componentRef + domain.fileSuffix);
+    Path parent = dir.toPath().resolve(getSubDirFor(componentRef));
+    try {
+      FileUtils.forceMkdir(parent.toFile());
```
Comment (dbmeneses) [inline sonar-scanner-protocol/src/main/java/org/sonar/scanner/protocol/output/FileStructure.java]: Files.createDirectories(parent)

**Comment by dbmeneses** [comment]
1 comment, then LGTM

**Comment by henryju** [comment]
@dbmeneses Good catch, fixed.

**Comment by sns-seb** [comment]
@Duarte I think we should rather stick with Java NIO and use https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#createDirectories(java.nio.file.Path,%20java.nio.file.attribute.FileAttribute...)

Commons IO is packed with performance issues and after Commons Collection, I think we should aim at removing this Commons library next

**Comment by henryju** [comment]
@sns-seb That was exactly the point of @dbmeneses feedback :)

**Comment by sns-seb** [comment]
true, I read it the otherway around, sorry

**Comment by duarte** [comment]
Hey, I think you @ mentioned the wrong duarte

**Pull request merged**